### PR TITLE
Add Deneb (EIP-4844) types and presets

### PR DESCRIPTION
**Pull request by jimmygchen**

### CODE REVIEW

File: src/kzg.rs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            expected: KZG_COMMITMENT_BYTES_LEN,
+        })?;
+        Ok(Self(inner))
+    }
+}
+
+#[derive(Default, Debug, Clone, SimpleSerialize, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", serde(into = "String", try_from = "String"))]
+pub struct KzgProof(ByteVector<KZG_PROOF_BYTES_LEN>);
```
Comment (ralexstokes) [inline src/kzg.rs:89]: I think for now we can just get away with `pub type KzgProof = ByteVector<KZG_PROOF_BYTES_LEN>;`

and drop the rest of the code defining/impl'ing `KzgProof`

File: src/kzg.rs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    #[error(
+        "invalid length of encoding: expected {expected} bytes but only provided {provided} bytes"
+    )]
+    EncodingError { provided: usize, expected: usize },
+}
+
+#[derive(Default, Debug, Clone, SimpleSerialize, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", serde(into = "String", try_from = "String"))]
+pub struct KzgCommitment(ByteVector<KZG_COMMITMENT_BYTES_LEN>);
```
Comment (ralexstokes) [inline src/kzg.rs:29]: I think for now we can just get away with `pub type KzgCommitment = ByteVector<KZG_COMMITmENT_BYTES_LEN>;`

and drop the rest of the code defining/impl'ing `KzgCommitment`

File: src/kzg.rs [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+#[derive(Debug, Error)]
+pub enum Error {
+    #[cfg(feature = "serde")]
+    #[error("error deserializing hex-encoded input: {0}")]
+    Hex(#[from] HexError),
+    #[error(
+        "invalid length of encoding: expected {expected} bytes but only provided {provided} bytes"
+    )]
+    EncodingError { provided: usize, expected: usize },
+}
```
Comment (ralexstokes) [inline src/kzg.rs:24]: do we need to keep this custom `Error` if we just use the simple type aliases below?

**Review by ralexstokes — COMMENTED** [review]
I'm going to focus on #168 first but I skimmed this

directionally, it looks good!

File: src/kzg.rs (hunk 1 above)
Comment (jimmygchen) [reply src/kzg.rs:89]: perfect, thanks!

File: src/kzg.rs (hunk 2 above)
Comment (jimmygchen) [reply src/kzg.rs:29]: 👍

File: src/kzg.rs (hunk 3 above)
Comment (jimmygchen) [reply src/kzg.rs:24]: Removed :) the less code the better

File: src/kzg.rs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+use crate::primitives::Bytes32;
+use crate::ssz::ByteVector;
+use ssz_rs::prelude::*;
+
+const KZG_COMMITMENT_BYTES_LEN: usize = 48;
+const KZG_PROOF_BYTES_LEN: usize = 48;
```
Comment (ralexstokes) [inline src/kzg.rs:6]: a bit random but may be worth making these `pub`, in the event some user of the library needed them for some reason...

Comment (jimmygchen) [reply src/kzg.rs]: Done. Make sense 👍

**Comment by jimmygchen** [comment]
Will revisit this PR once the spec changes for decoupling blobs (https://hackmd.io/cmYisgxkRuGe9NjX4gr97A?view) is closer to ready.

**Comment by jimmygchen** [comment]
Hi @ralexstokes 
I have made another update:
- replace all EIP-4844 references with Deneb
- implement latest changes from blobs decoupling (https://github.com/ethereum/consensus-specs/pull/3244)
- add blinded blob types

File: src/configs/sepolia.rs [hunk 5]
```diff
@@ -20,8 +20,8 @@ pub const BELLATRIX_FORK_VERSION: Version = [144, 0, 0, 113];
 pub const BELLATRIX_FORK_EPOCH: Epoch = 100;
 pub const CAPELLA_FORK_VERSION: Version = [3, 0, 16, 32];
 pub const CAPELLA_FORK_EPOCH: Epoch = FAR_FUTURE_EPOCH;
-pub const EIP4844_FORK_VERSION: Version = [4, 0, 16, 32];
-pub const EIP4844_FORK_EPOCH: Epoch = FAR_FUTURE_EPOCH;
+pub const DENEB_FORK_VERSION: Version = [4, 0, 16, 32];
+pub const DENEB_FORK_EPOCH: Epoch = FAR_FUTURE_EPOCH;
```
Comment (ralexstokes) [inline src/configs/sepolia.rs:24]: the `fork_version` looks incorrect (seems like the goerli values)

ill go fix in another PR and also set the sepolia fork epoch, since we know it now

**Review by ralexstokes — COMMENTED** [review]
nice updates!

File: src/configs/sepolia.rs (hunk 5 above)
Comment (jimmygchen) [reply src/configs/sepolia.rs:24]: I've created a PR to update the sepolia values for Capella
https://github.com/ralexstokes/ethereum-consensus/pull/194

Doesn't look like the Deneb config are out yet, so I'll leave them as it is.

File: src/deneb/beacon_block.rs [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    pub sync_aggregate: SyncAggregate<SYNC_COMMITTEE_SIZE>,
+    pub execution_payload: ExecutionPayload<
+        BYTES_PER_LOGS_BLOOM,
+        MAX_EXTRA_DATA_BYTES,
+        MAX_BYTES_PER_TRANSACTION,
+        MAX_TRANSACTIONS_PER_PAYLOAD,
+        MAX_WITHDRAWALS_PER_PAYLOAD,
+    >,
+    pub bls_to_execution_changes: List<SignedBlsToExecutionChange, MAX_BLS_TO_EXECUTION_CHANGES>,
+    pub blob_kzg_commitments: List<KzgCommitment, MAX_BLOBS_PER_BLOCK>,
```
Comment (jimmygchen) [inline src/deneb/beacon_block.rs:47]: May have to update `MAX_BLOBS_PER_BLOCK` to `MAX_BLOB_COMMITMENTS_PER_BLOCK`. 
https://github.com/ethereum/consensus-specs/pull/3338/files

**Comment by jimmygchen** [comment]
Hi @ralexstokes 
I've updated this PR to the latest spec - please take another look 🙏 
Cheers

**Review by ralexstokes — APPROVED** [review]
looks great!

I see what appear to be unnecessary generics on some of the blob types (e.g. params like `BYTES_PER_BLOB`)

that being said, I am doing some more extensive refactoring and want to just go ahead and merge this in to avoid excessive thrash from repo-wide change

if you run into some problem please feel free to open another PR
